examples_tests/44.LevelCurveExtraction/main.cpp

```diff
@@ -66,29 +66,80 @@ int main()
+    linePoints[outId+1u] = start.y;
+    linePoints[outId+2u] = start.z;
+    linePoints[outId+3u] = end.x;
+    linePoints[outId+4u] = end.y;
+    linePoints[outId+5u] = end.z;
+} 
+)===";
 
     //get the simple geometry shader data and turn it into ICPUSpecializedShader
-    auto shaderData = device->getAssetManager()->getFileSystem()->loadBuiltinData("irr/builtin/shaders/testGeomShader.geom");
+    auto len = strlen(geometryShaderCode);
+    auto shaderData = core::make_smart_refctd_ptr<asset::ICPUBuffer>(len);
+    memcpy(shaderData->getPointer(), geometryShaderCode, len);
     auto unspecializedShader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(shaderData), asset::ICPUShader::buffer_contains_glsl);
     auto shader = core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(unspecializedShader), asset::ISpecializedShader::SInfo({}, nullptr, "main", asset::ISpecializedShader::ESS_GEOMETRY));
     pipeline_cp->setShaderAtIndex(asset::ICPURenderpassIndependentPipeline::ESSI_GEOMETRY_SHADER_IX, shader.get());
 
-    //replace the ICPURenderpassIndependentPipeline with a copy that uses the replaced geometry shader
-    mesh_raw->getMeshBuffer(0u)->setPipeline(std::move(pipeline_cp));
+    mesh_raw->getMeshBuffer(0)->setPipeline(std::move(pipeline_cp));
+
 
     //create buffers for the geometry shader
-    auto lineCountBuffer = driver->createDeviceLocalGPUBufferOnDedMem(sizeof(uint32_t));
+    auto lineCountBuffer = driver->createDeviceLocalGPUBufferOnDedMem(roundUp(sizeof(irr::asset::DrawArraysIndirectCommand_t),16ull));
     uint32_t triangleCount;
     if (!asset::IMeshManipulator::getPolyCount(triangleCount, mesh_raw))
-        throw;
+        assert(false);
 
     auto linesBuffer = driver->createDeviceLocalGPUBufferOnDedMem(triangleCount * 6 * sizeof(float));
+    
+    driver->fillBuffer(lineCountBuffer.get(), offsetof(irr::asset::DrawArraysIndirectCommand_t, instanceCount), sizeof(uint32_t), triangleCount * 6 * sizeof(uint32_t));
+    //not sure what first refers to
+    driver->fillBuffer(lineCountBuffer.get(), offsetof(irr::asset::DrawArraysIndirectCommand_t, first), sizeof(uint32_t), triangleCount * 6 * sizeof(uint32_t));        
+    //not sure what baseInstance does
+    driver->fillBuffer(lineCountBuffer.get(), offsetof(irr::asset::DrawArraysIndirectCommand_t, baseInstance), sizeof(uint32_t), triangleCount * 6 * sizeof(uint32_t)); 
 
     const asset::SBufferBinding<video::IGPUBuffer> bufferBinding = { 0u ,linesBuffer };
-
     //we can safely assume that all meshbuffers within mesh loaded from OBJ has same DS1 layout (used for camera-specific data)
     //so we can create just one DS
-    asset::ICPUDescriptorSetLayout* ds1layout = mesh_raw->getMeshBuffer(0u)->getPipeline()->getLayout()->getDescriptorSetLayout(1u); //set 1u ---> 0u ?
+    asset::ICPUDescriptorSetLayout* ds1layout = mesh_raw->getMeshBuffer(0u)->getPipeline()->getLayout()->getDescriptorSetLayout(0u); //set 1u ---> 0u ?
```

Comment:
what's going on?
You have to create ds layout for things u declared in geom shader. Then you have to add it to pipeline layout of all pipelines being used. Ds layout is null while compiling shader because it really is null. Your pipeline layout doesn't have layout for ds0. See megatexture example, there's example of modifying pipelines of loaded meshes.
Then ofc you have to create desc set with those 2 SSBOs used by GS. I can help u with this on Monday